crates/subspace-networking/src/node_runner.rs

```diff
@@ -211,7 +211,7 @@ where
                     self.handle_peer_dialing().await;
 
                     self.peer_dialing_timeout =
-                        Box::pin(tokio::time::sleep(Duration::from_secs(3)).fuse());
+                        Box::pin(tokio::time::sleep(Duration::from_secs(15)).fuse());
```

Comment:
15 seconds is way too much IMHO. If it takes 15 seconds just to establish connection, there is no way it'll work well. I'm fine bumping it to 5 seconds though.

Why did you need so big timeout?

Reply:
We have keep_alive=15s for our inner protocols. Most of the time during the active phase we'll have multiple established connections and during the idle phase, we don't need to reconnect so often. I'm setting it to 5 for this PR. But I'll likely revisit it soon when we finalize our connectivity discussion.